src/main/java/com/google/api/codegen/transformer/java/JavaTestTransformer.java

```diff
@@ -86,11 +94,41 @@ private ApiTestClassView createTestClassView(Interface service) {
     testClass.apiClassName = surfaceNamer.getApiWrapperClassName(service);
     testClass.name = testClass.apiClassName + "Test";
     testClass.mockServiceClassName = "Mock" + testClass.apiClassName;
-    testClass.testCases = new ArrayList<>();
+    testClass.testCases = createTestCaseViews(service);
 
     String outputPath = pathMapper.getOutputPath(service, apiConfig);
     testClass.outputPath = outputPath + "/" + testClass.name + ".java";
     testClass.imports = typeTable.getImports();
     return testClass;
   }
+
+  private List<ApiTestCaseView> createTestCaseViews(Interface service) {
```

Comment:
In fact, you should go ahead and move this now - this function should be re-used by each language, but it's currently in a Java-specific transformer.